Source/RiveRenderer/Private/Platform/pls_render_context_rhi_impl.hpp

```diff
@@ -0,0 +1,240 @@
+#pragma once
+#include <chrono>
+
+#include "Shaders/ShaderPipelineManager.h"
+
+THIRD_PARTY_INCLUDES_START
+#undef PI
+#include "rive/renderer/rive_renderer.hpp"
+#include "rive/renderer/rive_render_image.hpp"
+#include "rive/renderer/buffer_ring.hpp"
+THIRD_PARTY_INCLUDES_END
+
+class RIVERENDERER_API RenderTargetRHI : public rive::gpu::RenderTarget
+{
+public:
+    RenderTargetRHI(FRHICommandList& RHICmdList, const FTexture2DRHIRef& InTextureTarget);
+    virtual ~RenderTargetRHI() override {}
+    
+    FTexture2DRHIRef texture()const
+    {return m_textureTarget;}
+
+    FUnorderedAccessViewRHIRef targetUAV()const
+    {return m_targetUAV;}
+
+    FUnorderedAccessViewRHIRef coverageUAV()const
+    {return m_coverageUAV;}
+
+    FUnorderedAccessViewRHIRef clipUAV()const
+    {return m_clipUAV;}
+
+    FUnorderedAccessViewRHIRef scratchColorUAV()const
+    {return m_scratchColorUAV;}
+    
+private:
+    FTexture2DRHIRef m_scratchColorTexture;
+    FTexture2DRHIRef m_textureTarget;
+    FTexture2DRHIRef m_atomicCoverageTexture;
+    FTexture2DRHIRef m_clipTexture;
+
+    FUnorderedAccessViewRHIRef m_coverageUAV;
+    FUnorderedAccessViewRHIRef m_clipUAV;
+    FUnorderedAccessViewRHIRef m_scratchColorUAV;
+    FUnorderedAccessViewRHIRef m_targetUAV;
+};
+
+class StructuredBufferRingRHIImpl;
+
+class BufferRingRHIImpl final : public rive::gpu::BufferRing
+{
+public:
+    BufferRingRHIImpl(EBufferUsageFlags flags, size_t in_sizeInBytes, size_t stride);
+    void Sync(FRHICommandList& commandList) const;
+    FBufferRHIRef contents()const;
+    
+protected:
+    virtual void* onMapBuffer(int bufferIdx, size_t mapSizeInBytes) override;
+    virtual void onUnmapAndSubmitBuffer(int bufferIdx, size_t mapSizeInBytes) override;
+    
+private:
+    FBufferRHIRef m_buffer;
+    EBufferUsageFlags m_flags;
+};
+
+template<typename UniformBufferType>
+class UniformBufferRHIImpl : public rive::gpu::BufferRing
+{
+public:
+    UniformBufferRHIImpl(size_t in_sizeInBytes): BufferRing(in_sizeInBytes)
+    {
+        //m_buffer =  TUniformBufferRef<UniformBufferType>::CreateEmptyUniformBufferImmediate( UniformBuffer_MultiFrame);
+    }
+
+    void Sync(FRHICommandList& commandList, int offset)
```

Comment:
`offset` should probably be a `size_t` if dealing when memory sizes.

And just for my own understanding, Is the size built into `UniformBuffer_SingleFrame`? How does this know how much data to sync?